### ISSUE TITLE
Write tests

### ISSUE THREAD

**Issue by btk** [btk/pre-require#12]
Create multiple test cases for normal usage, `.format` and `.search` methods using your choice of javascript testing tool;

Jest might be a good choice https://facebook.github.io/jest/docs/en/getting-started.html

**Comment by gatukgl** [comment]
I'm working on it. 🎉

**Comment by btk** [comment]
@gatukgl that's great, if you have any question with the usage cases you can ask in this issue.

**Comment by gatukgl** [comment]
@btk I wonder, to test `.format` and `.search` function. Is that ok to test on `assets.js` file?

**Comment by btk** [comment]
@gatukgl yes I think you should initilize your own `assets.js` file before starting your test. Like you can have a test asset folder;

```
test-assets/
 - asset1.png
 - asset2.png
 - ...
```
Thank you can initilize the `node index test-assets/ assets.js` 

Then you can use the assets.js file that you generated for your tests.

**Comment by gatukgl** [comment]
I just have an idea to separate both `search()` and `format()` in `index.js` to be real functions (not just a plain text) so we can easily write unit tests. Would you mind if I open a new issue to refactor that two. :D 

So it will looks like below.

```js
   // index.js
    ...

    let assetsFileContent = `// import this file for static assets
    let Obj = {
      ${slugArray.join(', ')},
      search: ${searchFile},
      format: ${searchFileFormat}
    }
    export default Obj;
    `;

   ...

const searchFile = function(key) {
  if(this.hasOwnProperty(key)) {
    return this[key];
  } else {
    return -1;
  }
}

const searchFileFormat = function(extension) {
  let filteredArray = this.filter(assetSlug => (assetSlug.includes("-"+extention)));
    if(filteredArray.length){
       return filteredArray;
    } else {
       return -1;
    }
}
```

**Comment by btk** [comment]
@gatukgl yeah, that is actually really smart!

How do you plan to get them functions into text so the generated file will be as intended? Your current code, (I'm not sure but) `${searchFile}` won't work? Maybe you can use something like `function.prototype.toString()`

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/toString

If you can manage the "real" functions to be printed in the assets.js as intended, I would love to see your pr. Thank you.

**Comment by gatukgl** [comment]
Great!! I’ll try :) Thank you.

**Comment by gatukgl** [comment]
@btk  I've already opened PR to test  `.search()` and I'm just curious, for the `.filter()` in `.format()`, does it work with an object?

**Comment by btk** [comment]
Handled by #31 

We might have a new issue to list other possible test cases to be implemented in future. Thank you.
